default.nix

```diff
@@ -452,20 +473,29 @@ let
           fi
 
           echo Installing JavaScript Dependencies
-          ${pkgs.yarn}/bin/yarn
-          echo Generating psc-package config
-          ${pkgs.yarn}/bin/yarn spago psc-package-insdhall
-          echo Installing PureScript Dependencies
-          ${pkgs.yarn}/bin/yarn psc-package install
-          echo Generating nix config
-          ${pp2n}/bin/pp2n psc-package2nix
-          ${yarn2nix.yarn2nix}/bin/yarn2nix > yarn.nix
-          cp .psc-package/local/.set/packages.json packages.json
+          yarn
+
+          echo Generating nix configs.
+          yarn2nix > yarn.nix
+          # spago2nix generate
+
           echo Done
         '';
       };
 
-      withDevTools = env: env.overrideAttrs (attrs: { nativeBuildInputs = attrs.nativeBuildInputs ++ [ packages.cabal-install pkgs.git pkgs.cacert ]; });
+      withDevTools = env: env.overrideAttrs (attrs: { nativeBuildInputs = attrs.nativeBuildInputs ++ 
+                                                                        [ packages.cabal-install 
+                                                                          pkgs.git 
+                                                                          pkgs.cacert 
+                                                                          pkgs.nodejs-10_x
+                                                                          pkgs.nodePackages_10_x.node-gyp
+                                                                          pkgs.yarn 
+                                                                          pkgs.yarn2nix
+                                                                          easyPS.purs
+                                                                          easyPS.psc-package
+                                                                          easyPS.spago
+                                                                          easyPS.spago2nix
```

Comment:
why all these additions?

Reply:
these are needed to run the same version of the purescript tooling and to update the dependencies. Some of these will go in a future PR when we neaten up this stuff.

Reply:
I'm down with having the dependencies we need so you can `nix-shell; psc` or whatever. Can the ones for updating the deps not stay in the update script?

Reply:
Anyway, can we at least have some comments about what things are needed for.

Reply:
I'll add some comments.

We need these in the shell because currently the update script doesn't work. I could remove it all together but then I would have to put it back in. It _almost_ works. Actually can you check that `nix-shell` still works for you, I am having trouble getting shell things to work on my nixos machine. `bash: /run/user/0/env-vars: Permission denied` for example.